src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health < 0) {
+      Animal.removeDead();
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - The Animal class's applyDamage method currently calls Animal.removeDead directly within the method when an animal's health goes below 0. This can be inefficient if multiple animals receive damage in a loop, as it would repeatedly filter the Animal.alive array. To optimize performance, consider removing dead animals from the Animal.alive array at a more appropriate time, such as after all damage has been applied, to avoid unnecessary filtering.